src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -152,6 +153,71 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 
 
 /*
+ * BuildDistinctPlan create an returns plan for distinct. Depending on
+ * availability of hash function it chooses HashAgg over Sort/Unique
+ * plans.
+ * This function has a potential performance issue since we blindly set
+ * Plan nodes without looking at cost. We might need to revisit this
+ * if we have performance issues with select distinct queries.
+ */
+static Plan *
+BuildDistinctPlan(Query *masterQuery, Plan *topLevelPlan)
+{
+	bool distinctClausesHashable = true;
+	List *distinctClauseList = masterQuery->distinctClause;
+	Plan *plan = topLevelPlan;
+	List *aggregateTargetList = copyObject(masterQuery->targetList);
+	List *aggregateColumnList = pull_var_clause_default((Node *) aggregateTargetList);
+	ListCell *columnCell = NULL;
+
+	Assert(masterQuery->distinctClause);
+	Assert(!masterQuery->hasDistinctOn);
+
+	/*
+	 * make_agg function requires that columns in aggregate target list
+	 * has valid varno which is OUTER_VAR in our case. We should
+	 * probably do this update earlier in multi_master_planner so
+	 * that aggregate generating function do not need to do this.
+	 */
+	foreach(columnCell, aggregateColumnList)
+	{
+		Var *column = (Var *) lfirst(columnCell);
+		column->varno = OUTER_VAR;
+	}
+
+	/*
+	 * Create group by plan with HashAggregate if all distinct
+	 * members are hashable, and query does not have distinct on
+	 * and query does not already have sort plan on distinct clauses.
+	 */
+	distinctClausesHashable = grouping_is_hashable(distinctClauseList);
+	if (distinctClausesHashable)
+	{
+		/* finally create the plan */
```

Comment:
we can drop this comment or move to before the `make_agg`

Reply:
removed the line